Propagate series tags to qBittorrent when adding torrent to download client

neBM opened the pull request:
QBittorrent client

#### Description

QBittorrent has support for Tags in its API. This PR is for allowing users to opt-in for tagging newly added torrents with the [Tags of the series](https://github.com/Sonarr/Sonarr/blob/4b50861a6b458f11b1cf4c173df83bf4efe1cd1e/src/NzbDrone.Core/Tv/Series.cs#L62).

#### Screenshots for UI Changes

Addition of "Propagate Tags" option.

![image](https://github.com/user-attachments/assets/7033e85d-54ee-4d23-ab30-83c2a1470390)

#### Issues Fixed or Closed by this PR
* Closes #7930

neBM committed:
Propagate series tags to qBittorrent when adding torrent to download client

neBM commented:
Currently trailing this PR on personal instance.

Will remove draft status after I can guarantee this works without issues.

markus101 commented on src/NzbDrone.Core/Localization/Core/en.json:
```diff
@@ -485,6 +485,8 @@
   "DownloadClientQbittorrentSettingsContentLayoutHelpText": "Whether to use qBittorrent's configured content layout, the original layout from the torrent or always create a subfolder (qBittorrent 4.3.2+)",
   "DownloadClientQbittorrentSettingsFirstAndLastFirst": "First and Last First",
   "DownloadClientQbittorrentSettingsFirstAndLastFirstHelpText": "Download first and last pieces first (qBittorrent 4.1.0+)",
+  "DownloadClientQbittorrentSettingsPropagateTags": "Propagate Tags",
```
I think `Add Series Tags` would be a better name.

```suggestion
  "DownloadClientQbittorrentSettingsAddSeriesTags": "Add Series Tags",
```

markus101 commented on src/NzbDrone.Core/Localization/Core/en.json:
```diff
@@ -485,6 +485,8 @@
   "DownloadClientQbittorrentSettingsFirstAndLastFirst": "First and Last First",
   "DownloadClientQbittorrentSettingsFirstAndLastFirstHelpText": "Download first and last pieces first (qBittorrent 4.1.0+)",
+  "DownloadClientQbittorrentSettingsPropagateTags": "Propagate Tags",
+  "DownloadClientQbittorrentSettingsPropagateTagsHelpText": "Add Season Tags to new torrents when added to the download client (qBittorrent 4.1.0+)",
```
In the context of Sonarr, series = show, not series = season so it'd be:

```suggestion
  "DownloadClientQbittorrentSettingsPropagateTagsHelpText": "Add series tags to new torrents when added to the download client (qBittorrent 4.1.0+)",
```

markus101 commented on src/NzbDrone.Core/Download/Clients/QBittorrent/QBittorrent.cs:
```diff
@@ -80,6 +84,7 @@ protected override string AddFromMagnetLink(RemoteEpisode remoteEpisode, string
             var isRecentEpisode = remoteEpisode.IsRecentEpisode();
             var moveToTop = (isRecentEpisode && Settings.RecentTvPriority == (int)QBittorrentPriority.First) || (!isRecentEpisode && Settings.OlderTvPriority == (int)QBittorrentPriority.First);
             var forceStart = (QBittorrentState)Settings.InitialState == QBittorrentState.ForceStart;
+            var propagateTags = Settings.PropagateTags;
```
No need to store this in a separate variable since it's only used once and is a simple boolean.

markus101 commented on src/NzbDrone.Core/Download/Clients/QBittorrent/QBittorrentSettings.cs:
```diff
@@ -73,6 +73,9 @@ public QBittorrentSettings()
         public int ContentLayout { get; set; }
 
+        [FieldDefinition(14, Label = "DownloadClientQbittorrentSettingsPropagateTags", Type = FieldType.Checkbox, HelpText = "DownloadClientQbittorrentSettingsPropagateTagsHelpText")]
+        public bool PropagateTags { get; set; }
```
To match the suggested naming.

```suggestion
        public bool AddSeriesTags { get; set; }
```

markus101 reviewed (changes requested):
Looks pretty good, mostly naming/wording.

stevietv reviewed (changes requested):
thanks for contributing this. Made a few style comments below.

neBM committed:
Resolve PR review comments

- Renamed "Propagate tags" et al to "Add Series Tags"
- Inlined PropagateTags transient variable
- Sorted JSON

neBM committed:
Resolve PR review comments

- Change casing in help text

stevietv commented on src/NzbDrone.Core/Localization/Core/en.json:
```diff
@@ -486,6 +486,8 @@
   "DownloadClientQbittorrentSettingsFirstAndLastFirst": "First and Last First",
   "DownloadClientQbittorrentSettingsFirstAndLastFirstHelpText": "Download first and last pieces first (qBittorrent 4.1.0+)",
   "DownloadClientQbittorrentSettingsInitialStateHelpText": "Initial state for torrents added to qBittorrent. Note that Forced Torrents do not abide by seed restrictions",
+  "DownloadClientQbittorrentSettingsAddSeriesTags": "Add Series Tags",
```
needs re-sorting now that the variables are changed

neBM committed:
Resolve PR review comments

- Sorted JSON (again oops)

neBM committed:
Add missing check for add series tags

neBM committed:
Only call add tags if there are tags to add

markus101 reviewed (approved):
Thanks for implementing this.
